Core/HLE/sceUsbMic.h

```diff
@@ -16,4 +16,61 @@
 // https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
 
 #pragma once
+
+#include "sceKernel.h"
+#include <mutex>
+
 void Register_sceUsbMic();
+
+void __UsbMicInit();
+void __UsbMicShutdown();
+void __UsbMicDoState(PointerWrap &p);
+
+struct MicWaitInfo {
+	SceUID threadID;
+	u32 addr;
+	u32 needSize;
+	u32 sampleRate;
+};
+
+class QueueBuf {
```

Comment:
We have a few similar queues around the code base - any particular reason none of them could be re-used? 

I do understand how it can be easiest to just make another one that does exactly what you need though, and I am fine with that, at least temporarily - just wondering if you looked around at the others and found something.

Reply:
I hope to have a queue buffer that can automatically increase its capacity, because the data is dynamic, which can ensure that the buffer capacity is sufficient and secure. However, I have not found any similar queue, or maybe I missed it?

Reply:
Let's merge and maybe look at this later.